### PR TITLE
[CR-305] harmonizing diff exp results (edgepy part)

### PR DESCRIPTION
**Pull request by EpigeneMax**
## Description and Context
See [CR-305]
This PR focuses on defining a common class for diff exp results and using it in the `edgepy` module.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] No code (non-breaking change that does not impact code: formatting, build system, documentation...)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Release (version number increment, possibly with documentation updates)

[CR-305]: https://epigenelabsteam.atlassian.net/browse/CR-305?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: inmoose/edgepy/DGEExact.py [hunk 1]
```diff
@@ -16,9 +16,28 @@
...
+    _metadata = ["comparison", "genes"]
+
+    @property
+    def _constructor(self):
+        def f(*args, **kwargs):
+            return DGEExact(
+                *args, comparison=self.comparison, genes=self.genes, **kwargs
+            )
+
+        return f
```
Comment (guillaumeap) [inline inmoose/edgepy/DGEExact.py:34]: Why is it necessary to specify that here?

**Review by guillaumeap — APPROVED** [review]
Looks good! I just have a question out of curiosity :)

File: inmoose/edgepy/DGEExact.py (hunk 1 above)
Comment (EpigeneMax) [reply inmoose/edgepy/DGEExact.py:34]: Basically, `DGEExact` is a `pandas.DataFrame` with 2 extra attributes (`comparison` and `genes`).
Its constructor (`__init__`) takes any arguments valid for `pandas.DataFrame.__init__` and forwards them to `pandas.DataFrame.__init__`, plus 2 arguments to initialize the 2 extra attributes.

The `_constructor` property returns the **constructor** to use when building a sub-frame _e.g._ through indexing, `.loc` or `.iloc`. The returned constructor is automatically passed the arguments used to build the sub-`pandas.DataFrame`, but here we want to pass it two extra arguments (for the 2 extra attributes).
This is what `f` is for: passing `self.comparison` and `self.genes` on to the sub-`DGEExact` being built.

For all the nitty-gritty details, have a look at [pandas documentation](https://pandas.pydata.org/docs/development/extending.html#subclassing-pandas-data-structures).

Comment (guillaumeap) [reply inmoose/edgepy/DGEExact.py:34]: Ok thanks!

**Pull request merged**